buildSrc/src/main/kotlin/dev/nolij/zumegradle/JarCompressing.kt

```diff
@@ -247,35 +224,44 @@ fun applyProguard(jar: File, minecraftConfigs: List<MinecraftConfig>, configDir:
 	proguardCommand.add("-libraryjars")
 	proguardCommand.add(libraries.joinToString(File.pathSeparator) { "\"$it\"" })
 
-	val configuration = Configuration()
-	ConfigurationParser(proguardCommand.toTypedArray(), System.getProperties())
-		.parse(configuration)
-
-	try {
-		ProGuard(configuration).execute()
-	} catch (ex: Exception) {
-		throw IllegalStateException("ProGuard failed for $jar", ex)
-	} finally {
-		inputJar.delete()
+	val mappings = MemoryMappingTree()
+
+	configDir.listFiles { _, name -> name != "common.pro" }?.forEach {
+		println("Running ProGuard pass ${it.nameWithoutExtension}")
+		val thisCommand = proguardCommand.toMutableList()
+		thisCommand.add(1, "@${it.absolutePath}")
+		
+		val configuration = Configuration()
+		ConfigurationParser(thisCommand.toTypedArray(), System.getProperties())
+			.parse(configuration)
+
+		try {
+			ProGuard(configuration).execute()
+			MappingReader.read(mappingFile.toPath(), MappingFormat.PROGUARD, mappings)
+		} catch (ex: Exception) {
+			throw IllegalStateException("ProGuard failed for $jar", ex)
+		} finally {
+			inputJar.delete()
+		}
+	}
+	
+	MappingWriter.create(mappingFile.toPath(), MappingFormat.TINY_2).use {
+		mappings.accept(it)
```

Comment:
What do the mappings produced by this code look like with multiple passes?

Reply:
you can check the mappings from CI, but it's just tiny2 mappings. if you're asking what multiple passes will do to it, I have no idea, but it might not be good.

Reply:
Yes, that's what I was asking.